packages/terra-application-layout/src/menu/_PrimaryNavigationSideMenu.jsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable import/no-extraneous-dependencies, import/no-webpack-loader-syntax, import/first, import/no-unresolved, import/extensions  */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter, matchPath } from 'react-router-dom';
+import NavigationSideMenu from 'terra-navigation-side-menu';
+import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
+import ApplicationLayoutPropTypes from '../utils/propTypes';
+
+const propTypes = {
+  /**
+   * The array of navigation Objects to be rendered as items within the navigation menu.
+   */
+  navigationItems: ApplicationLayoutPropTypes.navigationItemsPropType.isRequired,
+  /**
+   * The Object of layout-related APIs provided to the components of the Layout.
+   */
+  layoutConfig: ApplicationLayoutPropTypes.layoutConfigPropType.isRequired,
+  /**
+   * The Object containing RoutingStack APIs provided to children of the RoutingStack.
+   */
+  routingStackDelegate: RoutingStackDelegate.propType.isRequired,
+  /**
+   * The location from the router context. This prop is provided by the `withRouter` HOC-generator.
+   */
+  location: PropTypes.shape({
+    pathname: PropTypes.string,
+  }).isRequired,
+};
+
+/**
+ * This string is used as the root level menu identifier for the PrimaryNavigationSideMenu's side
+ * menu implementation.
+ */
+const navMenuKey = 'navigationMenu';
+
+class PrimaryNavigationSideMenu extends React.Component {
+  /**
+   * Builds and returns the NavigationSideMenu API items from the given navigationItems.
+   */
+  static buildChildSideNavItems(navigationItems) {
+    return navigationItems.map(item => ({
+      key: item.path,
+      text: item.text,
+      hasSubMenu: !!item.hasSubMenu,
+      metaData: {
+        /**
+         * metaData is added to the item to make handleMenuChange's life easier.
+         */
+        url: item.path,
+        hasSubMenu: !!item.hasSubMenu,
+      },
+    }));
+  }
+
+  /**
+   * Finds and returns the first navigationItem path that matches the current location.
+   */
+  static getSelectedChildKey(path, navigationItems) {
+    for (let i = 0, itemCount = navigationItems.length; i < itemCount; i += 1) {
+      const navItemPath = navigationItems[i].path;
+      if (matchPath(path, { path: navItemPath })) {
+        return navItemPath;
+      }
+    }
+    return undefined;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.handleMenuChange = this.handleMenuChange.bind(this);
+
+    this.state = {
+      selectedMenuKey: navMenuKey,
+      selectedChildKey: PrimaryNavigationSideMenu.getSelectedChildKey(props.location.pathname, props.navigationItems),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      selectedMenuKey: navMenuKey,
```

Comment:
When do we expect `componentWillRecieveProps`? If a consumer chooses to use the the app.show() without the parent show?